### PR TITLE
[update] Licence boilerplate

### PR DESCRIPTION
**Pull request by toxa81**
* adopt BSD 3-clause licence
* add list of authors
* change (simplify) cpp/hpp boilerplate

### CODE REVIEW

File: src/AUTHORS [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+Simon Pintarelli (simonpintarelli)
+Mathieu Taillefumier (mtaillefumier)
+Simon Frasch (AdhocMan)
+Harmen Stoppels (haampie)
+Ilia Sivkov (dithillobothrium)
+Man Hin Cheng (manhin321)
+Colin Kälin (ckae95)
+Rocco Meli (RMeli)
+Teodor Nikolov (teonnik)
+Tiziano Müller
```
Comment (RMeli) [inline src/AUTHORS]:
```suggestion
Tiziano Müller (dev-zero)
```

**Review by RMeli — COMMENTED** [review]
Just a couple of comments (feel free to ignore):
* The `AUTHORS` file might be better suited to the root directory
* DLA-Future uses a [license header check](https://github.com/eth-cscs/DLA-Future/blob/master/.github/workflows/check_license_header.yaml) which might be worth adding here as well

**Review by mtaillefumier — APPROVED** [review]
Everything is fine with me.

**Review by gsavva — APPROVED** [review]
all good for me

**Review by gcistaro — APPROVED** [review]
All good with me, thanks!

**Comment by toxa81** [comment]
@simonpintarelli @haampie @AdhocMan , please, have a look

**Review by simonpintarelli — APPROVED** [review]
lgtm

**Review by haampie — APPROVED** [review]

**Review by AdhocMan — APPROVED** [review]
lgtm

**Review by RMeli — APPROVED** [review]

**Pull request merged**